sway-lib-std/src/vec.sw

```diff
@@ -142,3 +145,84 @@ impl<T> Vec<T> {
         self.len == 0
     }
 }
+
+/// TODO: Add proper errors
+
+pub struct StorageVec<V> {}
+
+impl<V> StorageVec<V> {
+    #[storage(read, write)]
+    fn push(self, value: V) {
+        let len = get::<u64>(__get_storage_key());
+        let key = sha256((len, __get_storage_key()));
+        store::<V>(key, value);
+        store(__get_storage_key(), len + 1);
+    }
+
+    #[storage(read, write)]
+    fn pop(self) -> Option<V> {
+        let len = get::<u64>(__get_storage_key());
+        if len == 0 {
+            return Option::None;
+        }
+    
+        let key = sha256((len, __get_storage_key()));
+        store(__get_storage_key(), len - 1);
+
+        Option::Some(get::<V>(key))
+    }
+
+    #[storage(read)]
+    fn get(self, index: u64) -> Option<V> {
+        let len = get::<u64>(__get_storage_key());
+        if len <= index {
```

Comment:
For a user experience... do we want to continue to start indexing from 0 or from 1 @adlerjohn @otrho @nfurfaro ?

Reply:
Following Rust we should continue with 0, just like everything else we have. Is there a benefit to switching to 1 now?

Reply:
If there are no downsides to zero, then zero.